### ISSUE TITLE
Idea: Allow loading Plugins from secondary folder Data\Platform\[Folder]

### ISSUE THREAD

**Issue by mrowrpurr** [skyrim-multiplayer/skymp#552]
`tl;dr`
> If Plugins are read from 2 folders, we could use 1 folder for **distributing plugins** and the other folder for **developing plugins**.

---

### Scenario

As a Skyrim Platform developer
I can use Mod Organizer 2
And I can have Skyrim Platform plugins installed
And I can develop my own Skyrim Platform plugin
And I get hot reloading

### Steps

1. Install a mod which includes a Skyrim Platform plugin
   > _Or simply create an empty mod with a `Platform\Plugins` folder to reproduce the experience of downloading someone else's Skyrim Platform plugin_
2. Create a new plugin
3. Compile your plugin and deploy the .js file to `Skyrim Special Edition\Data\Platform\Plugins\[plugin].js`
4. Run the game
5. See your new plugin do something, e.g. print to the console
6. Recompile your plugin and overwrite the .js file `Skyrim Special Edition\Data\Platform\Plugins\[plugin].js`

### Expected

7. The code is hot reloaded and the change is seen in the game

### Actual

7. The code change does NOT come into effect and the change is NOT seen in the game

### Why?

If any mod contains a `Platform\Plugins` folder, then it will be mounted via MO2's `vfs`

Anything mounted with `vfs` cannot be updated at runtime

### How else is this problematic?

This severely restricts Skyrim Platform developers _who have other Skyrim Platform plugins installed_ or who are authoring plugins which work with other plugins.

Basically, if you want to write a Skyrim Platform plugin (_using hot reloading_), then you cannot have any Skyrim Platform plugins installed (_via MO2_).

That's not ideal.

### Potential Solution

Enable reading plugins from a secondary folder (_with any name_)

For example:
- `Platform\Plugins\**\*.js` is read and plugins are loaded from here <--- _no hot reloading required_
- `Platform\Dev\**\*.js` is read and plugins are loaded from here _and the files are watched and hot reloaded_

If Plugins are read from two folders, then we could use 1 folder for **distributing plugins** and the other folder for **developing folders**.

**Related:**
- https://github.com/skyrim-multiplayer/skymp/issues/421

**Comment by mrowrpurr** [comment]
@Pospelove 

#### Notes from Testing .dll

##### With MO2 but no installed plugins (_no vfs mounted `Platform` folders_)

- `Plugins`
  - [ ] Loads OK `<----` **My Plugins folder plugins don't load on startup, but they do on hot reload**
  - [x] Hot reloading works
- `PluginsDev`
  - [x] Loads OK
  - [x] Hot reloading works
  
  ##### With MO2 with installed plugins (_vfs mounted `Platform/Plugins`_)

- `Plugins`
  - [x] Loads OK
  - [x] Hot reloading does not work (_as expected_)
- `PluginsDev`
  - [x] Loads OK
  - [x] Hot reloading works
